### ISSUE TITLE
Clarification: Is a + / – button a sufficient accessible alternative for pinch zoom if users can't control the zoom focus area? (WCAG 2.5.1)

### ISSUE THREAD

**Issue by MeenaPatel17** [w3c/wcag#4692]
I'm reviewing compliance with WCAG 2.1 Success Criterion 2.5.1: Pointer Gestures, which requires that functionality using multipoint gestures (such as pinch-to-zoom) has a single-pointer alternative unless the gesture is essential.

The [Understanding document](https://www.w3.org/WAI/WCAG21/Understanding/pointer-gestures.html)
 gives an example of a conforming pattern: a map that supports pinch-to-zoom and also provides + / – controls.

> A website includes a map view that supports the pinch gesture to zoom into the map content. User interface controls offer the operation using plus and minus buttons to zoom in and out.

However, in many common implementations, the + / – buttons only zoom into the center of the visible map, and users have no way to control the zoom focus area without dragging or panning.

For some pointer device users, this alternative (i.e., + / – buttons that zoom into a fixed area) does not provide the same level of functionality that users without disabilities can achieve. Users can’t easily zoom into a specific part of the map.

If evaluated under WCAG 2.2, this might also fail SC 2.5.7: Dragging Movements if panning the map relies solely on drag gestures and no simple pointer alternative (e.g., arrow keys or buttons) is provided.

Under WCAG 2.1, only SC 2.5.1 applies - so for the scope of 2.5.1 can the + and - buttons alone truly meet the requirement for a single-pointer alternative that provides equivalent functionality?

**Comment by mbgower** [comment]
**Official Working Group Response**

As you note, the pinch-to-zoom action involves a multiple-touch gesture, and so rightly can be assessed under 2.5.1 Pointer Gestures.

The ability to pan a map can be separate from an ability to zoom a map. (In example, one does not need to zoom in order to pan; one does not need to pan as one zooms.) As you point out, panning is typically invoked with a dragging movement, and so an alternative would be required under 2.5.7 Dragging Movements, and failing to do so would cause a failure of WCAG 2.2.

The [Understanding document](https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html) gives an exact example of this need:
> A map allows users to drag the view of the map around, and the map has up/down/left/right buttons to move the view as well.

In the scenario you describe, if pan and zoom are combined into one action that requires both multi-touch gestures as well as dragging, then you can fail the combined function under 2.5.1 at WCAG 2.1.

However, if a user can separately reposition the map view using a drag with only one finger, then that function does not fail 2.5.1. Assuming there is also a keyboard method of panning the map view, such a panning interaction would not appear to fail WCAG 2.1. As you point out, this means in such a situation, equivalent function is not addressed for a user unable to invoke a dragging movement. That is one of the reasons 2.5.7 was introduced in 2.2.

**Comment by GreggVan** [comment]
Dragging with one finger may pass that requirement -- but remember you still need to be able to zoom and to pan using only the keyboard interface

**Comment by detlevhfischer** [comment]
In maps, there is often a (somewhat less convenient) alternative of zooming in to another region via repeated activations of the [-] button followed by zooming in via single taps/clicks on another area of the map, where that single tap/click zooms in. I guess this might be minimally acceptable (i.e., meet the normative text, which does not say anything about "equivalent" or "effort") as a single pointer alternative for 2.5.1. and 2.5.7.

**Comment by mbgower** [comment]
> Dragging with one finger may pass that requirement -- but remember you still need to be able to zoom and to pan using only the keyboard interface

@GreggVan the issue is about 2.5.1, which covers pointer interaction, not keyboard; but the draft working group response also specifically mentions keyboard as a consideration.

> Assuming there is also a keyboard method of panning the map view, such a panning interaction would not appear to fail WCAG 2.1.

**Comment by GreggVan** [comment]
@mbgower   Not sure I  understand your question comment but 

yes - you need to be able to do everything from they keyboard - including panning an zooming an image

G

**Comment by patrickhlauke** [comment]
yes, but keyboard is a 2.1.1 consideration. we're asking specifically here about satisfying the aspects of 2.5.1 ...

**Comment by mbgower** [comment]
@detlevhfischer 
> In maps, there is often a (somewhat less convenient) alternative of zooming in to another region via repeated activations of the [-] button followed by zooming in via single taps/clicks on another area of the map, where that single tap/click zooms in. I guess this might be minimally acceptable (i.e., meet the normative text, which does not say anything about "equivalent" or "effort") as a single pointer alternative for 2.5.1. and 2.5.7.

I agree that this would meet the requirement, which lists no qualitative considerations about the method.

What you describe is basically what I need to do reposition in applications such as Figma (although I just click on somewhere on the screen to establish my new 'target' area, not on a button). Note that modifier keys _can_ allow the speed of zooming and repositioning with a single pointer to be quite effective.

**Comment by mbgower** [comment]
Working group review period concluded. Changing the [drafted working group](https://github.com/w3c/wcag/issues/4692#issuecomment-3412535990) response to the official response and closing the issue.

**Comment by detlevhfischer** [comment]
This bit seems incorrect in the [proposed WG response](https://github.com/w3c/wcag/issues/4692#issuecomment-3412535990) (which BTW I found pretty hard to parse):

> Assuming there is also a keyboard method of panning the map view, such a panning interaction would not appear to fail WCAG 2.1. 

2.5.1 and 2.5.7 are not in any way dependent on there being a keyboard method. What is required is (a series of) single pointer activation(s). As I have commented, I believe the presence of [+] and [-] controls can meet 2.5.1. and 2.5.7 if (and only if) after zooming out, some area on the map can be touched to zoom in to some other part of the map, effectively affording an alternative to panning.

**Comment by patrickhlauke** [comment]
reopening this, as I agree with @detlevhfischer ... keyboard panning is orthogonal to the question.

* if the map allows single-finger panning, and has buttons to +/- zoom, then it *passes* 2.5.1 Pointer Gestures (as it doesn't require multi-pointer or path-based gestures) ... however
* if the map can only be panned by dragging the single finger, then it *fails* 2.5.7 Dragging Movement
* whether or not there are keyboard methods is only relevant to determine pass/fail for 2.1.1 and don't "absolve" panning only be possible with single pointer dragging
